addons/hw_drivers/controllers/driver.py

```diff
@@ -250,6 +263,51 @@ def __init__(self):
         super(Manager, self).__init__()
         self.load_drivers()
 
+    def check_certificate(self):
+        """
+        Check if the current certificate is up to date or not authenticated
+        """
+        server = get_odoo_server_url()
+        if server:
+            try:
+                path = Path('/etc/ssl/certs/nginx-selfsigned.crt')
+                if path.exists():
+                    with path.open('r') as f:
+                        cert = crypto.load_certificate(crypto.FILETYPE_PEM, f.read())
+                        cert_end_date = datetime.datetime.strptime(cert.get_notAfter().decode('utf-8'), "%Y%m%d%H%M%SZ") - datetime.timedelta(days=10)
+                        for key in cert.get_subject().get_components():
+                            if key[0] == b'CN':
+                                cn = key[1].decode('utf-8')
+                        if cn == 'OdooTempIoTBoxCertificate' or datetime.datetime.now() > cert_end_date:
+                            self.load_certificate()
+            except Exception:
+                self.load_certificate()
+
+    def load_certificate(self):
+        """
+        Send a request to Odoo with customer db_uuid and enterprise_code to get a true certificate
+        """
+        db_uuid = read_file_first_line('odoo-db-uuid.conf')
+        enterprise_code = read_file_first_line('odoo-enterprise-code.conf')
+        if db_uuid and enterprise_code:
+            url = 'https://www.odoo.com/odoo-enterprise/iot/x509'
+            data = {
+                'params': {
+                    'db_uuid': db_uuid,
+                    'enterprise_code': enterprise_code
+                }
+            }
+            urllib3.disable_warnings()
+            http = urllib3.PoolManager(cert_reqs='CERT_NONE')
+            response = http.request(
+                'POST',
+                url,
+                body = json.dumps(data).encode('utf8'),
+                headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+            )
+            result = json.loads(response.data.decode('utf8'))['result']
+            subprocess.check_call([get_resource_path('point_of_sale', 'tools/posbox/configuration/load_certificate.sh'), result['x509_pem'], result['private_key_pem'], result['subject_cn']])
```

Comment:
we probably should not do that if enterprise_code and db_uuid are wrong and "request" return an error or result doesn't contains any result['x509_pem'], result['private_key_pem'], result['subject_cn']